### PR TITLE
Fix macos bigsur compilation

### PR DESCRIPTION
**Pull request by kdiez**

### CODE REVIEW

File: vendor/CppQuickCheck-2018-03-28/include/cppqc/Generator.h [hunk 1]
```diff
@@ -364,7 +365,11 @@ void sampleShrinkOutput(const Generator<T> &g, std::ostream &out,
             T x = g.unGen(rng, i);
             std::vector<T> shr = g.shrink(x);
             if (randomized)
-                std::random_shuffle(shr.begin(), shr.end());
+            {
```
Comment (BardurArantsson) [inline vendor/CppQuickCheck-2018-03-28/include/cppqc/Generator.h:368]: I was wondering about this one, but it seems (per cppreference) that this overload should have been removed in C++17, but apparently the compilers aren't always as strict as they should. TIL.

File: src/z-form.cc [hunk 2]
```diff
@@ -556,7 +556,7 @@ unsigned int vstrnfmt(char *buf, unsigned int max, const char *fmt, va_list vp)
  */
 static char *vformat(const char *fmt, va_list vp)
 {
-	static char *format_buf = NULL;
+	static char *format_buf = (char *)NULL;
```
Comment (BardurArantsson) [inline src/z-form.cc:559]: Can you check if a plain "nullptr" without the cast is accepted by your compiler here?

File: README.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ ToME is a [rogue-like](https://en.wikipedia.org/wiki/Roguelike) game.
 
 ## Getting Started
 
+### Objective of this fork
+
+A few bugs have been corrected for the code to compile in macOS Big Sur (11.2)
+
```
Comment (BardurArantsson) [inline README.md:10]: You can omit this change since it wont be a fork once I merge the PR :)

**Comment by BardurArantsson** [comment]
Looks good, just curious about one change and wonder if you could try my suggestion out.

File: README.md (hunk 3 above)
Comment (kdiez) [reply README.md:10]: OK :)

File: src/z-form.cc (hunk 2 above)
Comment (kdiez) [reply src/z-form.cc:559]: Done, compiles OK, and I agreee that it's neater this way.

**Comment by kdiez** [comment]
Hey @BardurArantsson! A bit late, but I have taken some time to get back to this.
It should be ready to merge now, please let me know if there are any outstanding bits.
Thanks!

**Comment by kdiez** [comment]
BTW, I have now tested the code in macOS 12.1 as well, it compiles (and seems to run) correctly with lastest updates.

**Comment by BardurArantsson** [comment]
LGTM, but I think you forgot to remove the change to the readme :)

**Comment by kdiez** [comment]
It's done now, thanks!

**Comment by BardurArantsson** [comment]
Rebased, squashed and merged. Thanks!

**Pull request closed without merging**